src/predicates/intersects.jl

```diff
@@ -93,37 +93,116 @@ function intersects(g₁::Geometry{Dim,T}, g₂::Geometry{Dim,T}) where {Dim,T}
 
   # move towards the origin
   d = O - P
+
   while true
     P = minkowskipoint(g₁, g₂, d)
     if (P - O) ⋅ d < zero(T)
       return false
     end
     push!(points, P)
 
-    # line segment case
-    if length(points) == 2
-      B, A = points
-      AB = B - A
-      AO = O - A
-      d = AB × AO × AB
-    else # simplex case
-      C, B, A = points
-      AB = B - A
-      AC = C - A
-      AO = O - A
-      ABᵀ = AC × AB × AB
-      ACᵀ = AB × AC × AC
-      if ABᵀ ⋅ AO > zero(T)
-        popat!(points, 1) # pop C
-        d = ABᵀ
-      elseif ACᵀ ⋅ AO > zero(T)
-        popat!(points, 2) # pop B
-        d = ACᵀ
-      else
-        return true
-      end
+    d = gjk!(O, points)
+    isnothing(d) && return true
+  end
+end
+
+"""
+    gjk!(origin::Point{Dim,T}, points) where {Dim,T}
+
+Perform one iteration of the GJK algorithm.
+
+It returns `nothing` if the `Dim`-simplex represented by `points`
+contains the origin point. Otherwise, it returns a vector with
+the direction for searching the next point.
+
+If the simplex is complete, it removes one point from the set to
+make room for the next point. A complete simplex must have `Dim + 1` points. 
+
+See also [`intersects`](@ref).
+"""
+function gjk! end
+
+function gjk!(origin::Point{2,T}, points) where T
```

Comment:
let's rename the function argument `origin` to `O` to keep the variable names as similar as possible in the git diff.